### PR TITLE
#163388141 Enable article like and dislike

### PR DESCRIPTION
**Pull request by dev-jey**
#### What does this PR do?
It enables a user to like or dislike articles of their choice.
#### Description of Task to be completed?
- create a route for article liking
- create a route for article dislike
- write tests for the feature
#### How should this be manually tested?
1. Clone the repo
```
git clone https://github.com/andela/ah-the-phoenix.git
```
2. Change it into the current directory
```
cd ah-the-phoenix
```
3. Checkout to the Feature branch
```    
git checkout ft-like-dislike-article-163388141
```
4. Run the server
`python manage.py runserver`

5.   1. Like
          Send a `patch` request to the url below to like an article.
          Remember to append the article slug 
         `http://127.0.0.1:8000/api/v1/articles/<slug>/like/`
      2. Dislike
          Send a `patch` request to the url below to dislike an article.
           Remember to append the article slug 
         `http://127.0.0.1:8000/api/v1/articles/<slug>/dislike/`

#### Any background context you want to provide?
None
#### What are the relevant pivotal tracker stories?
[#163388141 ](https://www.pivotaltracker.com/story/show/163388141)

#### Screenshots
1. Like
screen shot 2019-02-19 at 18 29 21

2. Dislike
screen shot 2019-02-19 at 18 30 07

### CODE REVIEW

File: authors/apps/articles/views.py [hunk 1]
```diff
@@ -85,3 +86,55 @@ def destroy(self, request, pk=None):
...
+    serializer_class = ArticleSerializer
+    permissions = (IsAuthenticatedOrReadOnly,)
+
+    def partial_update(self, request, pk=None):
+        """Update likes field."""
+        try:
+            article = Article.objects.get(pk=pk)
+        except Exception:
+            raise NotFound("The article does not exist")
+        if article in Article.objects.filter(
```
Comment (sirghiny) [inline authors/apps/articles/views.py]: It'll be faster to just check if the user is in the article.disliked_by queryset, this applies in both viewsets.

File: authors/apps/articles/views.py [hunk 2]
```diff
@@ -85,3 +86,55 @@ def destroy(self, request, pk=None):
...
+    def partial_update(self, request, pk=None):
+        """Update dislikes field."""
+        try:
+            article = Article.objects.get(pk=pk)
+        except Exception:
+            raise NotFound("The article does not exist")
+        if article in Article.objects.filter(
+                liked_by=request.user):
+            article.liked_by.remove(request.user)
+        if article in Article.objects.filter(
```
Comment (asirvex) [inline authors/apps/articles/views.py]: Can checking if article is liked or disliked by a user be abstracted into a function? Looks like the same logic is repeated in the like and dislike viewsets

File: authors/apps/articles/views.py (hunk 1 above)
Comment (dev-jey) [reply authors/apps/articles/views.py]: ok...working on it

File: authors/apps/articles/views.py (hunk 2 above)
Comment (dev-jey) [reply authors/apps/articles/views.py]: ok..working on it

**Pull request merged**